server/edit_subscription.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-server/v6/model"
+
+	"github.com/mattermost/mattermost-plugin-confluence/server/config"
+	"github.com/mattermost/mattermost-plugin-confluence/server/serializer"
+	"github.com/mattermost/mattermost-plugin-confluence/server/service"
+	"github.com/mattermost/mattermost-plugin-confluence/server/utils/types"
+)
+
+const subscriptionEditSuccess = "Your subscription has been edited successfully."
+
+func (p *Plugin) handleEditChannelSubscription(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
```

Comment:
This method is too long, can we split functionality into different methods so it's easier to maintain? Also, it's much easier to unittest this if we split it.

Thank you!